crates/dbsp/src/trace/spine_async/mod.rs

```diff
@@ -144,6 +148,22 @@ where
             .iter()
             .chain(self.merging_batches.iter().flatten())
     }
+
+    /// Returns this slot's contribution to the merger's priority.
+    ///
+    /// Ordinarily this is the number of batches in the slot, but a slot with 1
+    /// batch has a score of 0 because no merges are possible. Otherwise we can
+    /// end up giving a spine with 3 batches each in a different slot a higher
+    /// priority (which can't do any merging) a higher priority than a spine
+    /// with 2 batches in a single slot (which can).
+    fn score(&self) -> usize {
+        let n = self.n_batches();
+        if n == 1 {
+            0
+        } else {
+            n
```

Comment:
I wonder if this should be n-1, meaning that the priority of a slot is the number of batches that can be eliminated by fully merging it.

Reply:
That would be equivalent, this could be rewritten as `n.saturating_sub(1)`.

Reply:
I pushed a commit that does that simplification.

Reply:
(which also passes the QA test)

Reply:
Looks good, thanks! I hope @gz can review as well.